src/lib.rs

```diff
@@ -82,14 +81,10 @@ impl DstService {
     }
 
     #[tracing::instrument(skip(self), level = "info")]
-    async fn stream_destination(
-        &self,
-        scheme: String,
-        name: String,
-    ) -> mpsc::Receiver<GrpcResult<Update>> {
-        let state = self.state.read().await.get(&Dst { scheme, name }).cloned();
+    async fn stream_destination(&self, dst: &Dst) -> mpsc::Receiver<GrpcResult<Update>> {
+        let state = self.state.read().await.get(dst).cloned();
         if let Some(mut state) = state {
-            tracing::info!("destination exists");
+            tracing::info!(?dst, "exists");
```

Comment:
the destination should be included already from the span generated by `instrument` — is it not present?